workspace/src/app/views/shared/RuleEditor/model/RuleEditorModel.tsx

```diff
@@ -1422,6 +1670,47 @@ export const RuleEditorModel = ({ children }: RuleEditorModelProps) => {
         }, 1);
     };
 
+    const addStickyNoteToCanvas = (stickyNote: string, color: string, reactFlowWrapper) => {
+        const reactFlowBounds = reactFlowWrapper.current?.getBoundingClientRect();
+        const nodeId = currentStickyContent.get("nodeId");
+        if (!nodeId) {
+            const position = (reactFlowInstance as OnLoadParams).project({
+                x: (reactFlowBounds.width - DEFAULT_NODE_WIDTH) / 2,
+                y: (reactFlowBounds.height - DEFAULT_NODE_HEIGHT) / 2,
+            });
+            addStickyNode(stickyNote, position, color);
+        } else {
+            changeStickyNodeStyle(nodeId, color);
+            changeStickyNoteContent(nodeId, stickyNote);
+        }
+    };
+
+    function stickyMenuButtons(stickyId, color, stickyNote) {
+        return (
+            <>
+                <IconButton
+                    data-test-id={"edit-sticky-note"}
+                    name="item-edit"
+                    text={t("RuleEditor.node.executionButtons.edit.tooltip")}
+                    onClick={() => {
+                        setCurrentStickyContent(
```

Comment:
As already mentioned above the "view" code for editing the sticky note should probably be somewhere else, e.g. a `StickyNoteMenu` component.